backend/functions/src/types/index.ts

```diff
@@ -24,9 +24,7 @@ export type Group = {
   createTime: Date
   updateTime: Date
   // TODO replace the below with the new template timestamps
-  addStudentEmailTimestamp: Date | null
-  checkInEmailTimestamp: Date | null
-  shareMatchEmailTimestamp: Date | null
+  templateTimestamps: { [key: string]: Date }
```

Comment:
Remember to remove the TODO comment above, otherwise might be confusing in future (other files too)